[Design2-232] Copy Modal component to DSCO package

levadadenys opened the pull request:
<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->

### [Design2-232] Copy Modal component to DSCO package

feat(modal): copied modal to dsco/modal

## Links
- jira ticket: [Design2-232](https://codedotorg.atlassian.net/browse/DESIGN2-232)

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

<!--
- spec: []()
- jira ticket: []()
-->

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

kelbyhawn reviewed (approved):
Looks good, just noticed one minor styling discrepancy with the top margin on the content `<p>`:

<img width="925" alt="Screenshot 2025-01-10 at 9 51 37 AM" src="https://github.com/user-attachments/assets/73ea1744-c0fd-4025-a15b-e1e7b92c96aa" />

levadadenys commented:
> Looks good, just noticed one minor styling discrepancy with the top margin on the content `<p>`:
> 
> <img alt="Screenshot 2025-01-10 at 9 51 37 AM" width="925" src="https://private-user-images.githubusercontent.com/9256643/402104085-73ea1744-c0fd-4025-a15b-e1e7b92c96aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3OTA5MDksIm5iZiI6MTczNjc5MDYwOSwicGF0aCI6Ii85MjU2NjQzLzQwMjEwNDA4NS03M2VhMTc0NC1jMGZkLTQwMjUtYTE1Yi1lMWU3YjkyYzk2YWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMTNUMTc1MDA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDZhNGUxZmU3MTVhNGVlYzQ4MWM0YTc1MjgwYzViZjM2NGVjNzZhOWNiMTg2ZTVjOGJlYzA5OGUyOTdhMDg3MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FTx28Vb8JJmrO7ADuxYiLycjDgDaFWTSz63dxSzAr-c">

Thank you very much for this catch! Fixed and updated!

![image](https://github.com/user-attachments/assets/1c2ce118-b110-4537-a893-c0d6242db4a5)
